Pet_Care_ai_agent_secured_with_asgardeo/agent/agent.py

```
@@ -0,0 +1,556 @@
+            args = tool_call['args']
+            
+            # Safety Check: If we are in Scenario D (Ambiguity) but LLM tried to call a tool anyway,
+            # we can intercept it here. But usually, the strong prompt prevents this.
+            
+            print(f"🛠️ Agent calling {tool_name} with {args}")
+            selected_tool = next((t for t in tools if t.name == tool_name), None)
+            if selected_tool:
+                try:
+                    res = await selected_tool.ainvoke(args)
+                    tool_outputs.append(ToolMessage(content=str(res), tool_call_id=tool_call['id'], name=tool_name))
+                except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)